lib/grp.gi

```diff
@@ -285,6 +314,22 @@ end );
 #T for example one that checks only small integers?)
 
 InstallMethod( IsNilpotentGroup,
+    "if group size can be computed and is a prime power",
+    [ IsGroup and CanComputeSize ], 25,
+    function ( G )
+    local s;
+
+    s := Size ( G );
+    if IsInt( s ) and IsPrimePowerInt( s ) then
+        SetIsPGroup( G, true );
+        SetPrimePGroup( G, Factors( s )[1] );
+        return true;
+    fi;
+    TryNextMethod();
+    end );
```

Comment:
I am somewhat wary of this method. Being able to compute the size is not the same as being able to do so "quickly" or "easily" (as in `CanEasilyComputePcgs`). So this could lead to unexpected slow downs.

But perhaps I am just too careful here... The true problem, in any case, is that with all these automatic deductions, the system becomes very hard to predict; seemingly innocent changes can cause major changes in behaviour, both good and bad ones *sigh*.

So, I am not opposed to that change, just... wary... :/ But probably we should just ignore me in that regard!